extensions/api/data-management/transferprocess/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/transform/DataRequestToDataRequestDtoTransformerTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.transform;
+
+import org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.model.DataRequestDto;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DataRequestToDataRequestDtoTransformerTest {
+    DataRequestTransformerTestData data = new DataRequestTransformerTestData();
+
+    DataRequestToDataRequestDtoTransformer transformer = new DataRequestToDataRequestDtoTransformer();
+
+    @Test
+    void getInputType() {
+        assertThat(transformer.getInputType()).isEqualTo(DataRequest.class);
```

Comment:
Yes but don't we want the test to break if the method signature were to change?

Reply:
Actually it wouldn't break the test, given the use of `var` in the test code and reflection. Any change to a "similar" class would cause the test not to fail.

Reply:
this:
```
    @Override
    public Class<TransferProcess> getInputType() {
        return TransferProcessDto.class;
    }
```
will break compilation, so the typing forces us to return either a `TransferProcess.class` or `null`, so the easiest check for the test would be to assert that the return value is not null

Reply:
Right, but the point is: if I were to change the class under test to `implements DtoTransformer<DataRequest, SomeDummyClass>` and change `getOutputType` to return `SomeDummyClass.class`, the test would still pass if SomeDummyClass has the same fields as `TransferProcessDto`. We do want the test to verify that the transformer does produce `TransferProcessDto` specifically.

Reply:
That would break the `transform` test, probably enough

Reply:
It wouldn't. The test only does this. Because of the use of `var` and reflection it would not break

```
 var result = transformer.transform(data.entity, data.context);
        assertThat(result)
                .usingRecursiveComparison()
                .isEqualTo(data.dto);
```

Reply:
It might sound a bit of a far fetched scenario, but it seems sensible to formally verify what are the documented expectations of the class. In addition, if we were to produce multiple DTOs with a hierarchy of transformers, say, such a scenario might just happen

Reply:
Ok, so in your assumption you mean, if we declare a `SomeDummyClass` that's completely equal to the `InputType` and we substitute this, then test won't break. It's a bit "forced" scenario.
Tests should test behavior, not definitions/configurations.
However seems like a matter of taste, so no problems in letting it like this.

Reply:
it's actually about the *output type*, so it's part of behavior